scripts/build-upgrade-image.sh

```diff
@@ -71,6 +71,49 @@ aptly publish repo -skip-contents -skip-signing upgrade-repository
 VERSION=$(dpkg -f "$(find debs/ -name 'delphix-entire-*' | head -n 1)" version)
 sed "s/@@VERSION@@/$VERSION/" <version.info.template >version.info
 
-tar -I pigz -cf "$APPLIANCE_VARIANT.upgrade.tar.gz" version.info -C upgrade-scripts . -C ~/.aptly .
+sha256sum version.info >SHA256SUMS
+for dir in upgrade-scripts ~/.aptly; do
+	find "$dir" -type f -printf "%P\n" |
+		(
+			cd "$dir"
+			xargs sha256sum
+		) >>SHA256SUMS
+done
+
+#
+# We can't sign the actual contents of the SHA256SUMS file, since that'd
+# result is a payload that's too large for the signature service to
+# consume. Thus, to ensure the payload is small enough for the signature
+# service, we only sign the sha256sum of the SHA256SUMS file.
+#
+echo "{\"data\": \"$(sha256sum SHA256SUMS)\"}" >request.payload
+
+if [[ -n "${DELPHIX_SIGNATURE_TOKEN:-}" ]] && [[ -n "${DELPHIX_SIGNATURE_URL:-}" ]]; then
+	#
+	# Here, we need to generate signature files for all of the appliance
+	# versions that'll be allowed to upgrade from, using this upgrade
+	# image. For now, we hardcode this to simply generate a signature for
+	# the "5.3" release, but eventually we'll need to update this code to
+	# support more versions (as more linux-based versions are released).
+	#
+	for signature_version in "5.3"; do
+		curl -s -S -f -H "Content-Type: application/json" \
+			-u "$DELPHIX_SIGNATURE_TOKEN" -d @request.payload \
+			"$DELPHIX_SIGNATURE_URL/upgrade/keyVersion/${signature_version}/sign" \
+			>request.response
+
+		jq -r .signature <request.response >signature.base64
+		base64 -d signature.base64 >"signature.${signature_version}"
+
+		rm request.response signature.base64
+	done
+fi
+
+tar -I pigz -cf "$APPLIANCE_VARIANT.upgrade.tar.gz" \
+	"$(ls signature.* 2>/dev/null)" \
+	SHA256SUMS \
```

Comment:
What's the use case for including `SHA256SUMS` file in an upgrade image?

Reply:
apart from checking that the signature file was generated by delphix, we also need to verify that the signed hashes match the software in the tarball. Both of these together satisfies the security needs.

Reply:
correct; my plan is to update the `unpack-image` script (in the delphix-platform repo) to run: `sha256sum -c SHA256SUMS` immediately after extracting the upgrade image, so we can verify the files contained in the image haven't been modified.